test/unit/readers/ddf1csv.js

```diff
@@ -0,0 +1,808 @@
+      "grouping":{},
+      "orderBy":"time"
+    };
+
+    it("'Expects that Ddf instance will be created for ddfcsvReader'", function (done) {
+
+      var dcReader;
+
+      spyOn(Ddf, 'default');
+
+      dcReader = new ddfcsvReader.default(defaultReaderParams);
+
+      expect(Ddf.default).toHaveBeenCalled();
+
+      done();
+
+    });
+
+    it("'Expects that ddfcsvReader using Ddf should call getFilesList method'", function (done) {
+
+      var dcReader;
+
+      spyOn(Ddf.default.prototype, 'getFilesList');
+
+      dcReader = new ddfcsvReader.default(defaultReaderParams);
+
+      expect(Ddf.default.prototype.getFilesList).toHaveBeenCalled();
+
+      done();
+
+    });
+
+    it("'Expects that ddfcsvReader using Ddf should process getFilesList correct if files were not found'", function (done) {
+
+      var dcReader;
+
+      spyOn(XMLHttpRequest.prototype, 'send').and.callFake(function() {
+        return false;
+      });
+
+      dcReader = new ddfcsvReader.default(defaultReaderParams);
+
+      dcReader.read(defaultQuery, lang).then(function(){
+
+        fail("ddf-csv Reader, 'read' was resolved");
+        done();
+
+      }, function(){
+
+        expect(dcReader.ddf.filesList.length).toEqual(0);
+        done();
+
+      });
+
+    });
+
+    it("'Expects that ddfcsvReader should process file with incorrect datapoints'", function (done) {
+      var data,
+        dcReader,
+        updatedQuery;
+
+      var filesDataCSVUpdated = filesDataCSV.slice();
+      var selectIndex = filesDataCSVUpdated.indexOf('/base/.data/ddf/ddf--gapminder_world/output/ddf/ddf--datapoints--child_mortality_0_5_year_olds_dying_per_1000_born--by--geo--time.csv');
+      filesDataCSVUpdated[selectIndex] = '/base/test/fixture/ddf--datapoints--child_mortality_0_5_year_olds_dying_per_1000_born--by--geo--time.csv';
+
+      spyOn(Ddf.default.prototype, 'getFilesList').and.callFake(function () {
+        return filesDataCSVUpdated;
+      });
+
+      updatedQuery = {
+        "select":["geo","time","child_mortality_0_5_year_olds_dying_per_1000_born"],
+        "where":{"geo.is--country":true,"time":[["1800","2015"]]},
+        "grouping":{},
+        "orderBy":"time"
+      };
+
+      dcReader = new ddfcsvReader.default(defaultReaderParams);
+      dcReader.read(updatedQuery, lang).then(function(){
+
+        data = dcReader.getData();
+        expect(data.length).toEqual(0);
+
+        done();
+
+      }, function(){
+
+        fail("ddf-csv Reader, 'read' was rejected");
+        done();
+
+      });
+
+    });
+
+    it("'Expects that ddfcsvReader should skip non CSV files'", function (done) {
+      var updatedReaderParams,
+        dcReader,
+        filesFromLength,
+        filesReadyLength;
+
+      updatedReaderParams = utils.extend({}, defaultReaderParams);
+      updatedReaderParams.path = filesDataHTML['get-files.html'];
+
+      // 5 - csv files
+      // 1 - json
+
+      filesFromLength = 5;
```

Comment:
If html source has non CSV files they should be skipped.
You can check source of 'getFilesList' function, it parse html for links that ends with 'csv'.